modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectRelationshipResourceImpl.java

```diff
@@ -297,15 +297,31 @@ private com.liferay.object.model.ObjectDefinition _getObjectDefinition2(
 			ObjectRelationship objectRelationship)
 		throws Exception {
 
-		com.liferay.object.model.ObjectDefinition objectDefinition =
-			_objectDefinitionLocalService.
-				fetchObjectDefinitionByExternalReferenceCode(
-					objectRelationship.
-						getObjectDefinitionExternalReferenceCode2(),
-					contextCompany.getCompanyId());
+		com.liferay.object.model.ObjectDefinition
+			serviceBuilderObjectDefinition2 =
+				_objectDefinitionLocalService.
+					fetchObjectDefinitionByExternalReferenceCode(
+						objectRelationship.
+							getObjectDefinitionExternalReferenceCode2(),
+						contextCompany.getCompanyId());
+
+		if (serviceBuilderObjectDefinition2 != null) {
+			return serviceBuilderObjectDefinition2;
+		}
+
+		long rootObjectDefinitionId = 0;
+
+		if (objectRelationship.getEdge()) {
+			com.liferay.object.model.ObjectDefinition
+				serviceBuilderObjectDefinition1 =
+					_objectDefinitionLocalService.
+						getObjectDefinitionByExternalReferenceCode(
+							objectRelationship.
+								getObjectDefinitionExternalReferenceCode1(),
+							contextCompany.getCompanyId());
 
-		if (objectDefinition != null) {
-			return objectDefinition;
+			rootObjectDefinitionId =
+				serviceBuilderObjectDefinition1.getRootObjectDefinitionId();
 		}
 
 		ObjectFolder defaultObjectFolder =
```

Comment:
@G-Sa I would reorder here since `defaultObjectFolder.getObjectFolderId()` is used before the `rootObjectDefinitionId`

```

ObjectFolder defaultObjectFolder =
	_objectFolderLocalService.getOrAddDefaultObjectFolder(
		contextCompany.getCompanyId());

long rootObjectDefinitionId = 0;

if (objectRelationship.getEdge()) {
	com.liferay.object.model.ObjectDefinition
		serviceBuilderObjectDefinition1 =
			_objectDefinitionLocalService.
				getObjectDefinitionByExternalReferenceCode(
					objectRelationship.
						getObjectDefinitionExternalReferenceCode1(),
					contextCompany.getCompanyId());

	rootObjectDefinitionId =
		serviceBuilderObjectDefinition1.getRootObjectDefinitionId();
}
```